src/openzaak/components/besluiten/models.py

```diff
@@ -170,6 +193,17 @@ def __str__(self):
     def save(self, *args, **kwargs):
         if not self.identificatie:
             self.identificatie = generate_unique_identification(self, "datum")
+
+        # save previous zaak for triggers
+        if self.pk:
```

Comment:
doesn't it make more sense to override `__init__` and store `self._previous_zaak = self.zaak`? You can optimize this by keeping the loose-fk underlying attributes and define a property

```py
@property
def previous_zaak(self):
    ...
```

to handle it, but I don't think keeping that information in sync in the DB will go well. I'm worried that will give all weird kind of problems to debug :thinking:

Reply:
the realization is changed to use `__init__` and `@property` instead of model fields